src/main/java/com/microsoft/sqlserver/jdbc/KerbCallback.java

```diff
@@ -0,0 +1,61 @@
+package com.microsoft.sqlserver.jdbc;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Properties;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+
+public class KerbCallback implements CallbackHandler {
+
+    private final SQLServerConnection con;
+    private String usernameRequested = null;
+
+    KerbCallback(SQLServerConnection con) {
+        this.con = con;
+    }
+
+    private static String getAnyOf(Callback callback, Properties properties, String... names)
+            throws UnsupportedCallbackException {
```

Comment:
use the [formatter ](https://github.com/Microsoft/mssql-jdbc/blob/master/mssql-jdbc_formatter.xml) on new code.